src/gt4py/next/program_processors/runners/dace/gtir_to_sdfg_concat_where.py

```diff
@@ -0,0 +1,425 @@
+        )
+        lower_bound = dace.symbolic.pystr_to_symbolic(
+            f"max({concat_dim_bound - 1}, {output_domain[concat_dim_index][1]})"
+        )
+        lower_domain.insert(concat_dim_index, (concat_dim, lower_bound, concat_dim_bound))
+    elif concat_dim not in upper.gt_type.dims:  # type: ignore[union-attr]
+        assert (
+            upper.gt_type.dims  # type: ignore[union-attr]
+            == [
+                *lower.gt_type.dims[0:concat_dim_index],  # type: ignore[union-attr]
+                *lower.gt_type.dims[concat_dim_index + 1 :],  # type: ignore[union-attr]
+            ]
+        )
+        upper, upper_desc = _make_concat_field_slice(
+            sdfg, state, upper, upper_desc, concat_dim, concat_dim_index, concat_dim_bound
+        )
+        upper_bound = dace.symbolic.pystr_to_symbolic(
+            f"min({concat_dim_bound + 1}, {output_domain[concat_dim_index][2]})"
+        )
+        upper_domain.insert(concat_dim_index, (concat_dim, concat_dim_bound, upper_bound))
+    elif isinstance(lower_desc, dace.data.Scalar) or (
+        len(lower.gt_type.dims) == 1 and len(output_domain) > 1  # type: ignore[union-attr]
+    ):
+        assert len(lower_domain) == 1 and lower_domain[0][0] == concat_dim
+        lower_domain = [
+            *output_domain[:concat_dim_index],
+            lower_domain[0],
+            *output_domain[concat_dim_index + 1 :],
+        ]
+        lower, lower_desc = _make_concat_scalar_broadcast(
+            sdfg, state, lower, lower_desc, lower_domain, concat_dim_index
+        )
+    elif isinstance(upper_desc, dace.data.Scalar) or (
+        len(upper.gt_type.dims) == 1 and len(output_domain) > 1  # type: ignore[union-attr]
+    ):
+        assert len(upper_domain) == 1 and upper_domain[0][0] == concat_dim
+        upper_domain = [
+            *output_domain[:concat_dim_index],
+            upper_domain[0],
+            *output_domain[concat_dim_index + 1 :],
+        ]
+        upper, upper_desc = _make_concat_scalar_broadcast(
+            sdfg, state, upper, upper_desc, upper_domain, concat_dim_index
+        )
+    elif lower.gt_type.dims != upper.gt_type.dims:  # type: ignore[union-attr]
+        raise NotImplementedError("concat_where on fields with different domain is not supported.")
+
+    # ensure that the arguments have the same domain as the concat result
+    assert all(ftype.dims == output_dims for ftype in (lower.gt_type, upper.gt_type))  # type: ignore[union-attr]
+
+    lower_range_0 = output_domain[concat_dim_index][1]
+    lower_range_1 = dace.symbolic.pystr_to_symbolic(
+        f"max({lower_range_0}, {lower_domain[concat_dim_index][2]})"
+    )
+    lower_range_size = lower_range_1 - lower_range_0
+
+    upper_range_1 = output_domain[concat_dim_index][2]
+    upper_range_0 = dace.symbolic.pystr_to_symbolic(
+        f"min({upper_range_1}, {upper_domain[concat_dim_index][1]})"
+    )
```

Comment:
```suggestion
    upper_range_0 = dace.symbolic.pystr_to_symbolic(
        f"min({upper_range_1}, {upper_domain[concat_dim_index][1]})"
    )
    upper_range_1 = output_domain[concat_dim_index][2]
```
Now they have the same order as above.

Reply:
I see it now.
I guess there was some tricky thinking involved to get that right.